Lib/test/test_asyncio/test_pools.py

```diff
@@ -0,0 +1,182 @@
+"""Tests for pools.py"""
+
+import asyncio
+import unittest
+
+from unittest import mock
+
+
+def tearDownModule():
+    asyncio.set_event_loop_policy(None)
+
+
+class AbstractPoolTests(unittest.TestCase):
+
+    def setUp(self):
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.loop)
+
+    def tearDown(self):
+        self.loop.close()
+        self.loop = None
+
+    def test_methods_not_implemented(self):
+        pool = asyncio.pools.AbstractPool()
+        with self.assertRaises(NotImplementedError):
+            self.loop.run_until_complete(pool.astart())
+
+        with self.assertRaises(NotImplementedError):
+            self.loop.run_until_complete(pool.aclose())
+
+        func = mock.Mock()
+        with self.assertRaises(NotImplementedError):
+            self.loop.run_until_complete(pool.run(func))
+        func.assert_not_called()
+
+    def test_context_manager_not_implemented(self):
+        async def pool_cm():
+            async with asyncio.pools.AbstractPool() as pool:
+                pass
+
+        with self.assertRaises(NotImplementedError):
+            self.loop.run_until_complete(pool_cm())
+
+
+class ThreadPoolTests(unittest.TestCase):
```

Comment:
Could the tests benefit from `IsolatedAsyncioTestCase`? I guess it would remove `run_until_complete` calls along with using `async def main` . Doc : https://docs.python.org/3/library/unittest.html#unittest.IsolatedAsyncioTestCase

Reply:
Yeah, that seems like it would be quite useful for reducing a decent chunk of the boilerplate. I'll test it locally when I get the chance later, assuming @1st1 and @asvetlov are on board with using `IsolatedAsyncioTestCase` for these tests.

Reply:
Based on my local experimentation so far, it seems be to working as intended and does provide a bit of a readability benefit in terms to reducing white noise. Just waiting on a +1 from @1st1 since this would be the first CPython regression test to use `IsolatedAsyncioTestCase` (outside of `unittest` of course).

Reply:
Update: I'd like to consider using `IsolatedAsyncioTestCase` in a separate PR, after the initial version is implemented since we're a bit short on time with the upcoming feature deadline for 3.9 beta.